hivemind/averaging/matchmaking.py

```diff
@@ -78,8 +81,16 @@ def __init__(
 
         self.current_leader: Optional[PeerID] = None  # iff i am a follower, this is a link to my current leader
         self.current_followers: Dict[PeerID, averaging_pb2.JoinRequest] = {}  # my current followers excluding myself
-        self.potential_leaders = PotentialLeaders(self.peer_id, averaging_expiration, target_group_size)
-        self.data_for_gather: Optional[bytes] = None
+        self.potential_leaders = PotentialLeaders(self.peer_id, min_matchmaking_time, target_group_size)
+        self.step_control: Optional[StepControl] = None
+
+    @contextlib.asynccontextmanager
+    async def looking_for_group(self, step_control: StepControl):
```

Comment:
Methods are generally verbs and participles are attributes/properties, is it safe to rename this to `enable_looking_for_group`?

Reply:
And maybe if this method is used once, it's okay to just inline it (look_for_group is not that complicated)

Reply:
Not the case, this is a contextmanager.

with looking_for_group(...):